### PR TITLE
Introduced .isBefore() and .isAfter() in model.Node and view.Node

### LINKED ISSUE #1365: Introduce `isBefore()` and `isAfter()` to view and model nodes
Similarly to what we have in `model.Position` and `view.Position`, we could use `model.Node#isBefore` and `view.Node#isBefore`. Same for `#isAfter`. Of course, the methods would compare two nodes and tell if one is before the other one in the tree. If nodes are in different trees, the methods would return `false` (just like in positions).

This might be useful for example in sorting. Now we have to create positions before items and compare positions. It's not as much of hassle but is also prone to errors (if you'll try to create a position before detached element (root) it will throw).

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced `#isBefore` and `#isAfter` in `model.Node` and `view.Node`. Additionally, `model.Node#is` and `view.Node#is` and `view.Node#getPath` were added. Closes #1365.

### CODE REVIEW

File: src/view/node.js [hunk 1]
```diff
@@ -231,6 +217,23 @@ export default class Node {
...
 	 * @returns {Boolean}
 	 */
+	is( type ) {
+		return type == 'node';
+	}
+
+	/**
+	 * Clones this node.
+	 *
+	 * @method #clone
```
Comment (oskarwrobel) [inline src/view/node.js]: Should be protected.

File: src/model/node.js [hunk 2]
```diff
@@ -273,6 +274,63 @@ export default class Node {
...
+			return false;
+		}
+
+		// Return `false` if it is impossible to compare nodes.
+		if ( this.root !== node.root ) {
+			return false;
+		}
+
+		// In other cases, just check if the `node` is before, and return the opposite.
+		return !this.isBefore( node );
```
Comment (oskarwrobel) [inline src/model/node.js]: clever :)

File: src/view/position.js [hunk 3]
```diff
@@ -243,61 +243,39 @@ export default class Position {
...
 			return 'same';
 		}
 
-		// If positions have same parent.
-		if ( this.parent === otherPosition.parent ) {
-			return this.offset - otherPosition.offset < 0 ? 'before' : 'after';
-		}
+		// Get path from root to position's parent element. "Indexify" the paths (change elements to indices).
+		const thisPath = this.getAncestors().map( element => element.index );
+		const otherPath = otherPosition.getAncestors().map( element => element.index );
```
Comment (oskarwrobel) [inline src/view/position.js]: WDYT about using just introduced `Node#getPath()`?

```javascript
const thisPath = this.parent.is( 'node' ) ? this.parent.getPath() : [];
const otherPath = otherPosition.parent.is( 'node' ) ? otherPosition.parent.getPath() : []
```
and shifting `null` from array won't be necessary.

**Comment by oskarwrobel** [comment]
I left 2 small comments but works and looks fine in general 👍

**Comment by scofalik** [comment]
Fixed according to comments.

**Pull request merged**